huaweicloud/resource_huaweicloud_rds_instance_v3.go

```diff
@@ -334,1128 +317,281 @@ func resourceRdsInstanceV3Create(d *schema.ResourceData, meta interface{}) error
-	pathParameters := map[string][]string{
-		"id": []string{"job_id"},
-	}
-	for key, path := range pathParameters {
-		value, err := navigateValue(result, path, nil)
-		if err != nil {
-			return nil, fmt.Errorf("Error retrieving async operation path parameter, err=%s", err)
+	nodes := make([]map[string]interface{}, len(instance.Nodes))
+	for i, v := range instance.Nodes {
+		nodes[i] = map[string]interface{}{
+			"id":                v.Id,
+			"name":              v.Name,
+			"role":              v.Role,
+			"status":            v.Status,
+			"availability_zone": v.AvailabilityZone,
 		}
-		data[key] = value
 	}
-
-	url, err := replaceVars(d, "jobs?id={id}", data)
-	if err != nil {
-		return nil, err
+	if err := d.Set("nodes", nodes); err != nil {
+		return fmt.Errorf("[DEBUG] Error saving nodes to RDS instance (%s): %s", instanceID, err)
 	}
-	url = client.ServiceURL(url)
-
-	return waitToFinish(
-		[]string{"Completed"},
-		[]string{"Running"},
-		timeout, 5*time.Second,
-		func() (interface{}, string, error) {
-			r := golangsdk.Result{}
-			_, r.Err = client.Get(url, &r.Body, &golangsdk.RequestOpts{
-				MoreHeaders: map[string]string{
-					"Content-Type": "application/json",
-					"X-Language":   "en-us",
-				}})
-			if r.Err != nil {
-				return nil, "", nil
-			}
-
-			status, err := navigateValue(r.Body, []string{"job", "status"}, nil)
-			if err != nil {
-				return nil, "", nil
-			}
-			return r.Body, status.(string), nil
-		},
-	)
-}
-
-func fetchRdsInstanceV3ByList(d *schema.ResourceData, client *golangsdk.ServiceClient) (interface{}, error) {
-	identity := map[string]interface{}{"id": d.Id()}
-
-	queryLink := "?id=" + identity["id"].(string)
 
-	link := client.ServiceURL("instances") + queryLink
+	d.Set("tags", tagsToMap(instance.Tags))
 
-	return findRdsInstanceV3ByList(client, link, identity)
-}
-
-func findRdsInstanceV3ByList(client *golangsdk.ServiceClient, link string, identity map[string]interface{}) (interface{}, error) {
-	r, err := sendRdsInstanceV3ListRequest(client, link)
-	if err != nil {
-		return nil, err
+	if ok, err := isEmptyValue(reflect.ValueOf(instance.Nodes[0].AvailabilityZone)); err != nil {
+		return fmt.Errorf("Error fetching availability zone form rds instance %s, err = %s", instanceID, err)
+	} else if ok {
+		return fmt.Errorf("Rds instance %s: availability zone not exist", instanceID)
 	}
-
-	for _, item := range r.([]interface{}) {
-		val := item.(map[string]interface{})
-
-		bingo := true
-		for k, v := range identity {
-			if val[k] != v {
-				bingo = false
-				break
-			}
+	az1 := instance.Nodes[0].AvailabilityZone
+	if strings.HasSuffix(d.Get("flavor").(string), ".ha") {
+		if ok, err := isEmptyValue(reflect.ValueOf(instance.Nodes[1].AvailabilityZone)); err != nil {
+			return fmt.Errorf("Error fetching availability zone form rds instance %s, err = %s", instanceID, err)
+		} else if ok {
+			return fmt.Errorf("Rds instance %s is Ha mode but number of availability zone get one", instanceID)
```

Comment:
please make the error message more clear

Reply:
Unnecessary logic and error message has been cleared.